### PR TITLE
Bundle fpcalc with distributables

### PR DESCRIPTION
**Pull request by samj1912**
# Summary

* This is a…
    * [ ] Bug fix
    * [x] Feature addition
    * [ ] Refactoring
    * [ ] Minor / simple change (like a typo)
    * [ ] Other
* **Describe this change in 1-2 sentences**:

# Problem

# Solution

We were not bundling fpcalc properly while packaging. This PR fixes it.

# Action

### CODE REVIEW

File: picard.spec [hunk 1]
```diff
@@ -28,9 +29,17 @@ def get_locale_messages():
...
 block_cipher = None
 data_files = get_locale_messages()
+os_name = platform.system()
+fpcalc_name = 'fpcalc'
+
+if os_name == 'Windows':
+    fpcalc_name += '.exe'
+
+if os.path.isfile(fpcalc_name):
+    data_files += [(fpcalc_name, fpcalc_name)]
```
Comment (zas) [inline picard.spec]: Not sure it such program should go in datas or binaries: https://pythonhosted.org/PyInstaller/spec-files.html#spec-file-operation
How does it handle file perms ? (on systems like macosx)

Comment (zas) [reply picard.spec]: Also second argument of the tuple is supposed to be a folder:
https://pyinstaller.readthedocs.io/en/stable/spec-files.html#adding-data-files
"The second specifies the name of the **folder** to contain the files at run-time."

File: picard.spec [hunk 2]
```diff
@@ -28,9 +29,17 @@ def get_locale_messages():
     return data_files
 
 
-
 block_cipher = None
 data_files = get_locale_messages()
+os_name = platform.system()
```
Comment (zas) [inline picard.spec]: Just to be picky i would just write things this way:

```python
if platform.system() == 'Windows':
  fpcalc_name = 'fpcalc.exe'
else:
  fpcalc_name = 'fpcalc'
```

I don't see the point of using string concatenation to "build" the file name, `os_name` would be useful if used more than once, which isn't the case here.

File: picard/util/__init__.py [hunk 3]
```diff
@@ -197,6 +197,9 @@ def find_executable(*executables):
         executables = [e + '.exe' for e in executables]
     paths = [os.path.dirname(sys.executable)] if sys.executable else []
     paths += os.environ.get('PATH', '').split(os.pathsep)
+    # This is for searching for executables bundled in packaged builds
+    if getattr(sys, 'frozen', False):
+        paths += sys._MEIPASS
```
Comment (zas) [inline picard/util/__init__.py]: What's `sys._MEIPASS` ? Perhaps add a link to doc or smt.

File: picard/acoustid/__init__.py [hunk 4]
```diff
@@ -34,7 +35,10 @@ def __init__(self):
         self._running = 0
         self._max_processes = 2
 
-        if not config.setting["acoustid_fpcalc"]:
+        # The second condition is checked because in case of a packaged build of picard
+        # the temp directory that pyinstaller decompresses picard into changes on every
+        # launch, thus we need to ignore the existing config values.
+        if not config.setting["acoustid_fpcalc"] or getattr(sys, 'frozen', False):
```
Comment (zas) [inline picard/acoustid/__init__.py]: What about using something more explicit than `getattr(sys, 'frozen', False)` like `is_pyinstaller()`

File: picard.spec (hunk 2 above)
Comment (samj1912) [reply picard.spec]: It is used in the OSX pr again, so I thought I'd just include it here.

**Pull request merged**